src/bootstrap/test.rs

```diff
@@ -911,7 +917,20 @@ impl Step for Compiletest {
             cmd.arg("--lldb-python-dir").arg(dir);
         }
 
-        cmd.args(&build.config.cmd.test_args());
+        // Get paths from cmd args
+        let paths = match build.config.cmd {
+            Subcommand::Test { ref paths, ..} => &paths[..],
+            _ => &[]
+        };
+
+        // Get test-args by striping suite path
+        let mut test_args: Vec<&str> = paths.iter().filter(|p| p.starts_with(suite_path) &&
+           p.is_file()).map(|p| p.strip_prefix(suite_path).unwrap().to_str().unwrap()).collect();
+
+        if test_args.is_empty() {
```

Comment:
Any additional test args passed the command line should rather be appended to the existing set of paths.

Reply:
Thanks! going to fix this ASAP